scripts/script-tag.js

```diff
@@ -147,16 +148,92 @@
                     span.html(`${localDeliveryDate}`)
                 }
 
+                // if the value of the metafield is not _NA_ then only making the date field visible
                 if (metafield.value !== '_NA_') {
                     hcpreorderShipsFrom.css('visibility', 'visible');
                 }
 
                 // will handle the click event on the pre order button
                 preorderButton.on("click", addToCart);
             }
+        } else {
+            // this part executes on all the page other than product page
+            // finding an input field with name tags and then iterating over the same
+            jQueryPreOrder("input[name='tags']").map(async function (index, element) {
+
+                // checking for Pre-Orde or Back-Order tag
+                if (jQueryPreOrder(element).val().includes('Pre-Order') || jQueryPreOrder(element).val().includes('Back-Order')) {
+                    // getting the current display variant id and current virtual product id
+                    const variantId = jQueryPreOrder(element).siblings("input[name='id']").val();
```

Comment:
I think we should assign jQueryPreOrder(element) to a variable and then use further

Reply:
Sir, can we name it `const tagInput = jQueryPreOrder(element)`? as the element is input with name tag.

Reply:
It should be close to what it stores variantIdInput

Reply:
Sir, it stores the input field having tags, `variantTagInput`, WDYT?

Reply:
LGTM

Reply:
Done Sir